tests/unit/base/test_rest_object.py

```diff
@@ -0,0 +1,337 @@
+import pickle
+
+import pytest
+
+import gitlab
+from gitlab import base
+from tests.unit import helpers
+from tests.unit.helpers import FakeManager  # noqa: F401, needed for _create_managers
+
+
+def test_instantiate(gl, fake_manager):
```

Comment:
JFYI, this file is not passing mypy.

Reply:
Are you sure @lmilbaum? Seems ok to me on main:

```
(.venv) nejc@laptop:~/repos/python-gitlab$ tox -re mypy
mypy: remove tox env folder /home/nejc/repos/python-gitlab/.tox/mypy
mypy: install_deps> pip install -r /home/nejc/repos/python-gitlab/requirements-lint.txt -e .
mypy: commands[0]> mypy
Success: no issues found in 219 source files
  mypy: OK (13.37=setup[12.21]+cmd[1.16] seconds)
  congratulations :) (13.40 seconds)
(.venv) nejc@laptop:~/repos/python-gitlab$ git show
commit 14e0f65a3ff05563df4977d792272f8444bf4312 (HEAD -> main, origin/main, origin/HEAD)
```

Reply:
```
(.venv) lmilbaum@Lioras-Air-2 python-gitlab % tox -e pre-commit                            
pre-commit: commands[0]> pre-commit run --all-files --show-diff-on-failure
black....................................................................Passed
flake8...................................................................Passed
isort....................................................................Passed
pylint...................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

tests/unit/base/test_rest_object.py:11: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:17: error: "_create_managers" of "RESTObject" does not return a value
tests/unit/base/test_rest_object.py:23: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:25: error: Argument 2 to "FakeObject" has incompatible type "List[str]"; expected "Dict[str, Any]"
tests/unit/base/test_rest_object.py:28: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:40: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:54: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:65: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:78: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:87: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:106: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:111: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:119: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:127: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:132: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:143: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:150: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:156: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:162: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:168: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:175: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:181: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:214: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:224: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:235: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:239: error: "pprint" of "RESTObject" does not return a value
tests/unit/base/test_rest_object.py:250: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:259: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:265: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:279: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:287: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:293: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:299: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:308: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:318: error: Function is missing a type annotation
tests/unit/base/test_rest_object.py:334: error: Function is missing a type annotation
tests/unit/base/test_rest_manager.py:5: error: Function is missing a type annotation
tests/unit/base/test_rest_manager.py:8: error: Incompatible types in assignment (expression has type "Type[object]", base class "RESTManager" defined the type as "Optional[Type[RESTObject]]")
tests/unit/base/test_rest_manager.py:14: error: Function is missing a type annotation
tests/unit/base/test_rest_manager.py:17: error: Incompatible types in assignment (expression has type "Type[object]", base class "RESTManager" defined the type as "Optional[Type[RESTObject]]")
tests/unit/base/test_rest_manager.py:24: error: Function is missing a type annotation
tests/unit/base/test_rest_manager.py:27: error: Incompatible types in assignment (expression has type "Type[object]", base class "RESTManager" defined the type as "Optional[Type[RESTObject]]")
Found 42 errors in 2 files (checked 219 source files)

rst ``code`` is two backticks............................................Passed
rst directives end with two colons.......................................Passed
rst ``inline code`` next to normal text..................................Passed
renovate-config-validator................................................Passed
pre-commit: exit 1 (8.54 seconds) /Users/lmilbaum/repos/github.com/lmilbaum/python-gitlab> pre-commit run --all-files --show-diff-on-failure pid=81633
  pre-commit: FAIL code 1 (8.56=setup[0.02]+cmd[8.54] seconds)
  evaluation failed :( (8.59 seconds)
```